server.go

```diff
@@ -77,6 +77,20 @@ func Listen(conn net.PacketConn, config *Config) (Listener, error) {
 	}, nil
 }
 
+func populateConfig(config *Config) *Config {
+	versions := config.Versions
+	if len(versions) == 0 {
+		versions = make([]protocol.VersionNumber, len(protocol.SupportedVersions))
+		copy(versions, protocol.SupportedVersions)
+	}
+
+	return &Config{
+		TLSConfig: config.TLSConfig,
```

Comment:
It's interesting that we copy the config, but not the TLS config. That means that we'll pick up runtime changes to the TLSConfig, but not to ours. I'm not sure that matters, just as a note.